### PR TITLE
AMDGPU: Add subtarget feature for global atomic fadd denormal support

### PR DESCRIPTION
**Pull request by arsenm**
Not sure what the behavior for gfx90a is. The SPG says it always flushes.
The instruction documentation says it does not.

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96443?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#96443** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96443?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#96442** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96442?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95930** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95930?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95929** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95929?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by rampitec** [comment]
It is worse than that. It behaves differently depending on where atomic is executed. There is no single answer if this instruction supports denorms or not.

**Comment by arsenm** [comment]
> It is worse than that. It behaves differently depending on where atomic is executed. There is no single answer if this instruction supports denorms or not.

That doesn't matter. The flat case that sometimes flushes is just a no. Flushing is never a guarantee, we only need to know a flush may happen

**Review by rampitec — APPROVED** [review]

File: llvm/lib/Target/AMDGPU/GCNSubtarget.h [hunk 1]
```diff
@@ -167,6 +167,7 @@ class GCNSubtarget final : public AMDGPUGenSubtargetInfo,
   bool HasAtomicFlatPkAdd16Insts = false;
   bool HasAtomicFaddRtnInsts = false;
   bool HasAtomicFaddNoRtnInsts = false;
+  bool HasAtomicMemoryAtomicFaddF32DenormalSupport = false;
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/GCNSubtarget.h:170]: What does "AtomicMemoryAtomic" mean?

**Comment by arsenm** [comment]
### Merge activity

* **Jul 10, 8:37 AM EDT**: @arsenm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/96443).

**Pull request merged**